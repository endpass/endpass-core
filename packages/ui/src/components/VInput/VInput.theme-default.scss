.v-input-atom.theme-default:hover {
  border-color: $grey-2;

  .v-input-atom-control::placeholder {
    color: $grey-6;
  }
}

.v-input-atom.theme-default.is-focus {
  border-color: $primary;
  background-color: $white;

  .v-input-atom-control::placeholder {
    opacity: 0;
  }
}

.v-input-atom.theme-default.is-error {
  border-color: $error;
}

.v-input-atom.theme-default.is-disabled {
  border-color: $grey-1;

  .v-input-atom-control::placeholder {
    color: $grey-4;
  }
}

.v-input-atom.theme-default .v-input-atom-error.theme-default {
  color: $white;
  background-color: $error;
  font-size: 12px;
  width: 14px;
  min-width: 14px;
  height: 14px;
  border-radius: 50%;
  text-align: center;
  font-style: normal;
  margin-right: 17px;
  letter-spacing: 0;
}
