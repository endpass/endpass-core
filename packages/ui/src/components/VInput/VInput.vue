<template>
  <field-atom>
    <label-atom
      v-if="label"
      :class="{ 'is-muted': $attrs.disabled }"
      :label="label"
    />
    <description-atom
      v-if="description"
      :class="{ 'is-muted': $attrs.disabled }"
      :description="description"
    />
    <input-atom
      :value="value"
      :class="inputAtomCssClass"
      v-bind="$attrs"
      v-on="listeners"
      @focus="isFocus = true"
      @blur="isFocus = false"
    >
      <icon-atom
        v-if="error"
        class="v-input-atom-error"
      >
        !
      </icon-atom>
    </input-atom>
    <error-atom
      v-if="error"
      :error="error"
    />
  </field-atom>
</template>

<script>
import ThemeMixin from '@/mixins/ThemeMixin';
import LabelAtom from '@/atom/label-atom/label-atom';
import IconAtom from '@/atom/icon-atom/icon-atom';
import InputAtom from '@/atom/input-atom/input-atom';
import FieldAtom from '@/atom/field-atom/field-atom';
import ErrorAtom from '@/atom/error-atom/error-atom';
import DescriptionAtom from '@/atom/description-atom/description-atom';

export default {
  name: 'VInput',
  inheritAttrs: false,
  props: {
    label: {
      type: String,
      default: null,
    },
    value: {
      type: String,
      default: null,
    },
    error: {
      type: String,
      default: null,
    },
    description: {
      type: String,
      default: null,
    },
  },
  data() {
    return {
      isFocus: false,
    };
  },
  computed: {
    listeners() {
      return Object.assign(this.$listeners, {
        input: (value) => {
          this.$emit('input', value);
        },
      });
    },
    inputAtomCssClass() {
      return {
        'is-focus': this.isFocus,
        'is-error': this.error,
      };
    },
  },
  mixins: [ThemeMixin],
  components: {
    IconAtom,
    LabelAtom,
    InputAtom,
    FieldAtom,
    ErrorAtom,
    DescriptionAtom,
  },
};
</script>
