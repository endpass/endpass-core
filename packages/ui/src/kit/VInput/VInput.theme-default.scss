.label-atom.theme-default.is-muted {
  color: $grey-4;
}

.input-atom.theme-default:hover {
  border-color: $grey-2;

  .input-atom-control::placeholder {
    color: $grey-6;
  }
}

.input-atom.theme-default.is-focus {
  border-color: $primary-7;
  background-color: $white;

  .input-atom-control::placeholder {
    opacity: 0;
  }
}

.input-atom.theme-default.is-error {
  border-color: $error;
}

.input-atom.theme-default.is-disabled {
  border-color: $grey-1;

  .input-atom-control::placeholder {
    color: $grey-4;
  }
}

.input-atom.theme-default .icon-atom-error.theme-default {
  color: $white;
  background-color: $error;
  font-size: 12px;
  line-height: 14px;
  width: 14px;
  min-width: 14px;
  height: 14px;
  border-radius: 50%;
  text-align: center;
  font-style: normal;
  margin-right: 17px;
  letter-spacing: 0;
}
