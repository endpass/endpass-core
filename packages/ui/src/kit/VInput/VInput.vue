<template>
  <field-atom>
    <label-atom
      v-if="label"
      v-bind="$attrs"
      :label="label"
    />
    <description-atom
      v-if="description"
      v-bind="$attrs"
      :description="description"
    />
    <input-atom
      :value="value"
      :is-error="isError"
      v-bind="$attrs"
      v-on="listeners"
    >
      <icon-atom
        v-if="isError"
        :is-error="isError"
      >
        <svg-atom name="error" />
      </icon-atom>
    </input-atom>
    <error-atom
      v-if="isError"
      :error="error"
    />
  </field-atom>
</template>

<script>
import LabelAtom from '@/atom/label-atom/label-atom';
import InputAtom from '@/atom/input-atom/input-atom';
import FieldAtom from '@/atom/field-atom/field-atom';
import ErrorAtom from '@/atom/error-atom/error-atom';
import DescriptionAtom from '@/atom/description-atom/description-atom';
import IconAtom from '@/atom/icon-atom/icon-atom';
import SvgAtom from '@/atom/svg-atom/svg-atom';

export default {
  name: 'VInput',
  inheritAttrs: false,
  props: {
    label: {
      type: String,
      default: null,
    },
    value: {
      type: [String, Number],
      default: null,
    },
    error: {
      type: String,
      default: null,
    },
    description: {
      type: String,
      default: null,
    },
  },
  data() {
    return {
      isFocus: false,
    };
  },
  computed: {
    listeners() {
      return Object.assign(
        {},
        {
          ...this.$listeners,
          input: (value) => {
            this.$emit('input', value);
          },
        },
      );
    },
    isError() {
      return !!this.error;
    },
  },
  components: {
    LabelAtom,
    InputAtom,
    FieldAtom,
    ErrorAtom,
    DescriptionAtom,
    IconAtom,
    SvgAtom,
  },
};
</script>
