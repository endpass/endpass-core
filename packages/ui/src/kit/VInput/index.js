export { default } from './VInput.vue';
