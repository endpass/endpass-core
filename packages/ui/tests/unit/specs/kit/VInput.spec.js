import { shallowMount } from '@vue/test-utils';

import VInput from '@/kit/VInput';

describe('VInput', () => {
  let wrapper;

  describe('render', () => {
    const attrs = {
      id: 'id',
      placeholder: 'placeholder',
      disabled: 'disabled',
    };

    beforeEach(() => {
      wrapper = shallowMount(VInput, {
        attrs,
      });
    });

    it('should be a Vue component', () => {
      expect(wrapper.name()).toBe('VInput');
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should correctly render component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render component with label', () => {
      wrapper.setProps({ label: 'Label' });

      expect(wrapper.find('label-atom-stub').props().label).toBe('Label');
      expect(wrapper.find('label-atom-stub').exists()).toBe(true);
    });
  });

  describe('behavior', () => {
    it('should set default listener for input event', () => {
      wrapper = shallowMount(VInput, {
        attrs: {
          value: '',
        },
      });

      wrapper.find('input-atom-stub').vm.$emit('input', 'foo');

      expect(wrapper.emitted().input).toEqual([['foo']]);
    });
  });
});
