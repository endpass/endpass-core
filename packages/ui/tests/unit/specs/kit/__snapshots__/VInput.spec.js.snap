// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VInput render should correctly render component 1`] = `
<field-atom-stub>
  <!---->
   
  <!---->
   
  <input-atom-stub
    disabled="disabled"
    id="id"
    placeholder="placeholder"
  >
    <!---->
  </input-atom-stub>
   
  <!---->
</field-atom-stub>
`;
